curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64ff24b80431f62ec6b93f65.md

```diff
@@ -14,7 +14,16 @@ Remove the `myTaskArr` array and all of the code for `localStorage` because you
 You should remove `myTaskArr` and all the code related to `localStorage` that you've just learned.
 
 ```js
-assert.notMatch(code, /const\s+myTaskArr\s*=\s*\[\s*\{\s*task:\s('|")Walk\s*the\s*Dog\1\s*,\s*date:\s*('|")22-04-2022\2\s*\}\s*,\s*\{\s*task:\s('|")Read\s*some\s*books\3\s*,\s*date:\s*('|")02-11-2023\4\s*\}\s*,\s*\{\s*task:\s('|")Watch\s*football\5\s*,\s*date:\s*('|")10-08-2021\6\s*\}\s*,\s*\]\s*;?\s*localStorage\.setItem\(('|")data\7\s*,\s*JSON\.stringify\(\s*myTaskArr\s*\)\s*\)\s*;?\s*localStorage\.clear\(\s*\)\s*;?\s*const\s+getTaskArr\s*=\s*localStorage\.getItem\(\s*('|")data\8\s*\)\s*console\.log\(\s*getTaskArr\s*\)\s*const\s+getTaskArrObj\s*=\s*JSON\.parse\(\s*localStorage\.getItem\(\s*('|")data\9\s*\)\s*\)\s*;?\s*console\.log\(\s*getTaskArrObj\s*\)\s*;?/)
+assert.notMatch(code, /const\s+myTaskArr\s*=\s*\[\s*\{\s*task:\s*('|")Walk\s*the\s*Dog\1\s*,\s*date:\s*('|")22-04-2022\2\s*\}\s*,\s*\{\s*task:\s*('|")Read\s*some\s*books\3\s*,\s*date:\s*('|")02-11-2023\4\s*\}\s*,\s*\{\s*task:\s*('|")Watch\s*football\5\s*,\s*date:\s*('|")10-08-2021\6\s*\}\s*\]\s*;?\s*localStorage\.setItem\(('|")data\7\s*,\s*JSON\.stringify\(\s*myTaskArr\s*\)\s*;?\)\s*;?\s*localStorage\.clear\(\s*\)\s*;?\s*const\s+getTaskArr\s*=\s*localStorage\.getItem\(\s*('|")data\8\s*\)\s*console\.log\(\s*getTaskArr\s*\)\s*const\s+getTaskArrObj\s*=\s*JSON\.parse\(\s*localStorage\.getItem\(\s*('|")data\9\s*\)\s*\)\s*;?\s*console\.log\(\s*getTaskArrObj\s*\)\s*;?/);
+
+// ensures no remaining references to `myTaskArr` anywhere in the code
+assert.notMatch(code, /myTaskArr/);
+
+// ensures no reference to localStorage item "data" exists
+assert.notMatch(code, /localStorage\.getItem\(\s*('|")data\1\s*\)/);
+
+// ensures no reference to localStorage setItem for "data" exists
+assert.notMatch(code, /localStorage\.setItem\(\s*('|")data\1\s*,\s*JSON\.stringify\(\s*myTaskArr\s*\)\s*\)\s*;?/);
```

Comment:
Could you make each of these into separate test? And split the remaining, not covered yet, part of original regex into separate tests?

Reply:
sorry I did not get this. I might be lacking in certain context. Can you please explain this comment a bit more?

Reply:
hey @gikf any input on the above?

Reply:
@gikf  BTW I have made them into separate tests. Can you please review again and check if that's what you were talking about?